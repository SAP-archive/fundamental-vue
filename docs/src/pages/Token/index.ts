import { ExampleCollectionFunction } from "../types";

export const plugin: ExampleCollectionFunction = () => {
  return { status: "stable", icon: "tags", related: ["FdToken"] };
};
