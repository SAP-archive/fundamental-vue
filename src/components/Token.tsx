import { Component } from 'vue-property-decorator';
import { Api } from '@/api';
import { componentName } from '@/util';
import TsxComponent from '@/vue-tsx';

@Component({ name: componentName('Token') })
@Api.Component('Token')
@Api.Event('click', 'triggers when clicked')
@Api.defaultSlot('Token Text')
export class Token extends TsxComponent<{}> {
  public render() {
    return (
      <span
        class='fd-token'
        role='button'
        on-click={() => this.$emit('click', this)}
      >
        {this.$slots.default}
      </span>
    );
  }
}
