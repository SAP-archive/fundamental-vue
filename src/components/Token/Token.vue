<template>
  <span class="fd-token" role="button" v-on="$listeners">
    <slot />
  </span>
</template>

<script>
export default {
  name: "FdToken"
};
</script>
