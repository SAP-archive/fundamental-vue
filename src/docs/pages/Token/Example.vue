<title>Default Token</title>

<template>
  <div class="tokens">
    <FdToken @click="click">Click Me</FdToken>
    <FdToken>Lorem</FdToken>
    <FdToken>Dolor</FdToken>
    <FdToken>Filter</FdToken>
  </div>
</template>

<script>
export default {
  methods: {
    click() {
      window.alert("Clicked!");
    }
  }
};
</script>

<style scoped>
.tokens,
* {
  margin-right: 10px;
}
</style>
