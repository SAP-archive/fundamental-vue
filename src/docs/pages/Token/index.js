export default {
  relatedComponents: ["fd-token"]
};
