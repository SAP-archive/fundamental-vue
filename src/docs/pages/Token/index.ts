import { ExampleCollectionFunction } from '../types';

export const plugin: ExampleCollectionFunction = ({ Token }) => {
  return { icon: 'tags', relatedComponents: [Token] };
};
