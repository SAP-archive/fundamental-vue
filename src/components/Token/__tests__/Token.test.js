import { mount } from "@vue/test-utils";
import Token from "../Token.vue";

describe("Token", () => {
  const onClick = jest.fn();
  const text = "Token";
  const wrapper = mount(Token, {
    listeners: {
      click: onClick
    },
    slots: {
      default: text
    }
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("renders default slot when passed", () => {
    expect(wrapper.text()).toEqual(text);
  });

  it("calls onClick when token is clicked", () => {
    wrapper.trigger("click");
    expect(onClick).toHaveBeenCalled();
  });
});
