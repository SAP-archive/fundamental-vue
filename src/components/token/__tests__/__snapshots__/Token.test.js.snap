// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Token renders correctly 1`] = `
<span
  class="fd-token"
  role="button"
>
  Token
</span>
`;
